### PR TITLE
Fix version checker issues

### PR DESCRIPTION
**Pull request by xoriole**
- Fixes https://github.com/Tribler/tribler/issues/5678 : Referenced before assignment
- Fixes https://github.com/Tribler/tribler/issues/5670 : Multiple version checker popups

### CODE REVIEW

**Comment by devos50** [comment]
Shouldn’t this go to the 7.5 release branch?

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by devos50** [comment]
Ok, this is getting a bit problematic. We cannot easily run the "Clone Tribler" job since there is this annoying file with a unicode filename. However, the "Clone Tribler" job seems to be required for several of our checkers.

Given all the activity on the `release-7.5` branch, this is getting a priority. I will look into it.

**Comment by xoriole** [comment]
>Ok, this is getting a bit problematic. We cannot easily run the "Clone Tribler" job since there is this annoying file with a unicode filename. However, the "Clone Tribler" job seems to be required for several of our checkers.
>Given all the recent activity on the release-7.5 branch, this is getting a priority. I will look into it.

@devos50 Thanks

**Comment by devos50** [comment]
Please rebase this PR 👍

File: src/tribler-core/tribler_core/modules/tests/test_versioncheck.py [hunk 1]
```diff
@@ -116,6 +120,19 @@ async def test_version_check_timeout(self):
         self.should_call_new_version_callback = False
         await self.check_version()
 
+    @timeout(10)
+    async def test_version_check_api_timeout(self):
+        await self.setup_version_server(json.dumps({'name': 'v1337.0'}))
+
+        # Setting the API response lag to 2 seconds and
+        # client's version checkout timeout of 1ms, version checks should fail returning False
```
Comment (devos50) [inline src/tribler-core/tribler_core/modules/tests/test_versioncheck.py:128]: Is the checkout timeout in seconds or milliseconds (you are using both units)?

Comment (xoriole) [reply src/tribler-core/tribler_core/modules/tests/test_versioncheck.py:128]: Thanks for pointing out. It is in seconds. I'll update the comment.

**Review by devos50 — APPROVED** [review]

**Pull request merged**